janusgraph-core/src/main/java/org/janusgraph/graphdb/query/QueryProcessor.java

```diff
@@ -196,7 +196,7 @@ private LimitAdjustingIterator(BackendQueryHolder<B> backendQueryHolder) {
 
         @Override
         public Iterator<R> getNewIterator(int newLimit) {
-            if (!backendQuery.hasLimit() || newLimit>backendQuery.getLimit())
+            if (newLimit>backendQuery.getLimit())
```

Comment:
hasLimit() returns false iff limit == Integer.MAX_VALUE. If there is no limit, it is not only redundant but also incorrect from profiling standpoint, to update the limit (which will effectively return a new query).